### DISCUSSION TITLE
Support server side verification

### DISCUSSION THREAD

**Discussion by BrianTV98** [BrianTV98/slider_captcha#27] [Ideas]
**I received issue 26 about server side validation.**
**However, I don't really understand this feature yet. So I look forward to receiving analysis from the community**

**Comment by BrianTV98** [comment]
@i5hi

**Reply by i5hi** [reply]
FWIU of CAPTCHA one of the main purpose is to protect an API from spam - especially for unauthenticated routes like register - which would involve sending an email or phone OTP. So, the API server must provide the challenge and also verify it before sending mail or phone OTP.

Purely client side challenge can protect from abuse of a client but it does not protect an API from spam.

Sites like Binance do this for the register/login API so that users cannot spam the API using custom clients like a curl script which can trigger an API to send 1000's of emails/OTP to random addresses or phone number.

Does this make sense?
